utils/index.d.ts

```diff
@@ -2100,3 +2123,223 @@ export declare namespace validationUtils {
+    showInputBox<OptionsBaseT extends IAzureQuickPickOptions>(options: AgentInputBoxOptions<OptionsBaseT>): Promise<string>;
+
+    showWarningMessage<T extends MessageItem>(message: string, ...items: T[]): Promise<T>;
+    showWarningMessage<T extends MessageItem>(message: string, options: IAzureMessageOptions, ...items: T[]): Promise<T>;
+    showOpenDialog(options: AzExtOpenDialogOptions): Promise<Uri[]>;
+}
+
+// #endregion
+
+// #region Agent integration types
+
+/**
+ * A config that describes a command that the extension implements which makes use of wizards that use
+ * an {@link IAzureAgentInput}/{@link IAzureUserInput} to get user input.
+ */
+export type WizardBasedCommandConfig = {
+    type: "wizard";
+
+    /**
+     * A camel cased string that names the command.
+     * @example "createNewFunctionProject"
+     */
+    name: string;
+
+    /**
+     * The VS Code command ID that this command maps to.
+     * @example "azureFunctions.createNewFunctionProject"
+     */
+    commandId: string;
+
+    /**
+     * The display name of the command.
+     * @example "Create New Function Project"
+     */
+    displayName: string;
+
+    /**
+     * A sentence description that helps a LLM understand when the command should be used.
+     *
+     * The description should give an understanding of what a user prompt which matches to this
+     * command would look like. Give examples of terminology that the user might use, the type of
+     * statements they might make, and the type of questions they might ask. Also consider giving
+     * examples of what terminology or types of statements would not match to this command.
+     *
+     * For example:
+     *
+     * *This is best when users ask to create a Function App resource in Azure. They may refer
+     * to a Function App as 'Function App', 'function', 'function resource', 'function app
+     * resource', 'function app' etc. This command is not useful if the user is asking how to do something, or
+     * if something is possible.*
+     */
+    intentDescription?: string;
+
+    /**
+     * If the command requires that a workspace is currently open.
+     */
+    requiresWorkspaceOpen?: boolean;
+
+    /**
+     * If the command requires that the user is logged into Azure.
+     */
+    requiresAzureLogin?: boolean;
+};
+
+/**
+ * Information that should be available on the package.json of an extension which is compabitible with the Azure agent.
+ * This information should be placed in an `agentMetdata` property.
+ */
+export type ExtensionAgentMetadata = {
+    version: "1.0";
+
+    /**
+     * The VS Code command ID of a command that the extension implements which can be used to get the list
+     * of {@link WizardBasedCommandConfig}s that the extension implements and wishes to expose via the agent.
+     *
+     * Note: the function type is only for testing in the agent extension itself.
```

Comment:
I feel you have removed the function type here.

Reply:
Thanks!